cypress-io/cypress
# fix: respect CYPRESS_NO_COMMAND_LOG in env

- Closes https://github.com/cypress-io/cypress/issues/21721
- Closes https://github.com/cypress-io/cypress/issues/21271

### User facing changelog

The `CYPRESS_NO_COMMAND_LOG` environment variable will hide the command log when set to `1`. This is documented [here](https://docs.cypress.io/guides/references/troubleshooting#Disable-the-Command-Log) but hasn't been working correctly for a few major versions (left an empty space) and was not yet implemented in Cy 10.x.

### Additional details

This PR makes a `hideCommandLog` boolean available under `window.__CYPRESS_CONFIG`, since we want this variable accessible on the client side at startup time in run mode, which does not use GraphQL. 

In order to prevent rendering the reporter, we avoid rendering the element it would mount into. This let me find a few places where in 10.x, we assume it will always be there. So I added some checks where needed, and uncoupled finding the reporter element, which is allowed to fail, from finding the runner element, which should still throw if it cannot be found.

The tests include some (near!) duplication to cover run mode as well as open mode. I thought about making a custom command but for just 2 instances of a very short test, it didn't seem worth it.

### Steps to test

- Set `CYPRESS_NO_COMMAND_LOG=1` before any Cypress `run` or `open` command and open a spec.
- Verify that you don't see the Command Log or Inline Specs List with this setting enabled.
- Verify that in E2E and CT mode that the Application/Component Under Test is correctly centered and scale behavior is good. The AUT area should never overflow the window no matter how small.
- In open mode, the navigation bar on the left should be present and collapsed, in run mode, is should not be there at all.

### How has the user experience changed?

Cypress can run tests without the reporter again:

![Screen Shot 2022-06-30 at 7 11 27 PM](https://user-images.githubusercontent.com/8340719/176793389-6df199bc-85f2-4370-864c-a425c1bc06ef.png)


### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)?
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
